examples/counter-ts/core/reducers/counter.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  DECREMENT_COUNTER,
+  INCREMENT_COUNTER,
+} from "../constants";
+
+export interface CounterInitialState {
+  value: number;
+}
+
+const INITIAL_STATE = {
+  value: 0,
+};
+
+export default function counter(state: CounterInitialState = INITIAL_STATE, action: { type: string }): any {
```

Comment:
Instead of having constants defined in a separate file, it would be nice if the CounterAction would be exported, including the constants it allows:

```
export enum CounterActionType {
  DECREMENT_COUNTER,
  INCREMENT_COUNTER,
}
export interface CounterAction {
  type: CounterActionType
}
```